Source/ablastr/fields/EffectivePotentialPoissonSolver.H

```diff
@@ -0,0 +1,274 @@
+/* Copyright 2024 The WarpX Community
+ *
+ * This file is part of WarpX.
+ *
+ * Authors: Roelof Groenewald (TAE Technologies)
+ *
+ * License: BSD-3-Clause-LBNL
+ */
+/*
+ * This file was copied and edited from PoissonSolver.H in the same directory.
+ */
+#ifndef ABLASTR_EFFECTIVE_POTENTIAL_POISSON_SOLVER_H
+#define ABLASTR_EFFECTIVE_POTENTIAL_POISSON_SOLVER_H
+
+#include <ablastr/constant.H>
+#include <ablastr/utils/Communication.H>
+#include <ablastr/utils/TextMsg.H>
+#include <ablastr/warn_manager/WarnManager.H>
+#include <ablastr/math/fft/AnyFFT.H>
+#include <ablastr/fields/Interpolate.H>
+#include <ablastr/profiler/ProfilerWrapper.H>
+#include "PoissonSolver.H"
+
+#if defined(WARPX_USE_FFT) && defined(WARPX_DIM_3D)
+#include <ablastr/fields/IntegratedGreenFunctionSolver.H>
+#endif
+
+#include <AMReX_Array.H>
+#include <AMReX_Array4.H>
+#include <AMReX_BLassert.H>
+#include <AMReX_Box.H>
+#include <AMReX_BoxArray.H>
+#include <AMReX_Config.H>
+#include <AMReX_DistributionMapping.H>
+#include <AMReX_FArrayBox.H>
+#include <AMReX_FabArray.H>
+#include <AMReX_Geometry.H>
+#include <AMReX_GpuControl.H>
+#include <AMReX_GpuLaunch.H>
+#include <AMReX_GpuQualifiers.H>
+#include <AMReX_IndexType.H>
+#include <AMReX_IntVect.H>
+#include <AMReX_LO_BCTYPES.H>
+#include <AMReX_MFIter.H>
+#include <AMReX_MFInterp_C.H>
+#include <AMReX_MLMG.H>
+#include <AMReX_MLLinOp.H>
+#include <AMReX_MLNodeLaplacian.H>
+#include <AMReX_MultiFab.H>
+#include <AMReX_Parser.H>
+#include <AMReX_REAL.H>
+#include <AMReX_SPACE.H>
+#include <AMReX_Vector.H>
+#include <AMReX_MLEBNodeFDLaplacian.H>
+#ifdef AMREX_USE_EB
+#   include <AMReX_EBFabFactory.H>
+#endif
+
+#include <array>
+#include <optional>
+
+
+namespace ablastr::fields {
+
+/** Compute the potential `phi` by solving the Poisson equation with a modifed dielectric function
+ *
+ * Uses `rho` as a source. This uses the AMReX solver.
+ *
+ * More specifically, this solves the equation
+ * \f[
+ *   \nabla \cdot \sigma \nabla \phi = - \rho/\epsilon_0
+ * \f]
+ *
+ * \tparam T_PostPhiCalculationFunctor a calculation per level directly after phi was calculated
+ * \tparam T_BoundaryHandler handler for boundary conditions, for example @see ElectrostaticSolver::PoissonBoundaryHandler
+ * \tparam T_FArrayBoxFactory usually nothing or an amrex::EBFArrayBoxFactory (EB ONLY)
+ * \param[in] rho The charge density a given species
+ * \param[out] phi The potential to be computed by this function
+ * \param[in] sigma The semi-implicit matrix
```

Comment:
Could you update this to remove the mention of `semi-implicit`?

Reply:
```suggestion
 * \param[in] sigma The matrix representing the mass operator used to lower the local plasma frequency 
```